web/client/epics/geostory.js

```diff
@@ -461,6 +476,81 @@ export const handlePendingGeoStoryChanges = action$ =>
                     )
             )
     );
+
+/**
+ * Handle the url updates on currentPage change
+ * @param {Observable} action$ stream of actions
+ * @param {object} store redux store
+ */
+export const urlUpdateOnScroll = (action$, {getState}) =>
+    action$.ofType(UPDATE_CURRENT_PAGE)
+        .debounceTime(50) // little delay if too many UPDATE_CURRENT_PAGE actions come
+        .switchMap(({sectionId, columnId}) => {
+            if (
+                modeSelector(getState()) !== 'edit' && (!!columnId || !!sectionId)
+                && updateUrlOnScrollSelector(getState())
+            ) {
+                const EMPTY = 'EMPTY';
+                const storyId = get(getState(), 'geostory.resource.id');
+                const storyIds = window?.location?.hash?.substring(window?.location?.hash.indexOf(storyId)).split('/');
+                if (sectionId && storyId) {
+                    if (storyIds.length > 1 && storyIds[1]) {
+                            history.pushState(null, '', replace(window.location.href, storyIds.length === 3 ? `${storyIds[1]}/${storyIds[2]}` : `${storyIds[1]}`, `${sectionId}`));
+                    } else {
+                        if (window?.location?.hash?.includes('shared')) {
+                            history.pushState(null, '', `${window.location.href}${sectionId}`);
+                        } else {
+                            history.pushState(null, '', `${window.location.href}/${sectionId}`);
+                        }
+                    }
+                } else if (!sectionId && columnId && isString(columnId) && columnId !== EMPTY) {
+                    if (storyIds.length > 1) {
+                        if (window?.location?.hash?.includes('shared') && !storyIds[1]) {
+                            history.pushState(null, '',  `${window.location.href}`);
+                        } else {
+                            history.pushState(null, '', storyIds.length === 3 ? replace(window.location.href,  `${storyIds[2]}`, `${columnId}`) : `${window.location.href}/${columnId}`);
+                        }
+                    }
```

Comment:
Is it possible to reduce the nesting of this `if` statements or at least add some comments explaining the different `history.pushState` actions?

Reply:
done